src/server/api/routers/resource.ts

```diff
@@ -344,7 +366,7 @@ export const resourceRouter = createTRPCRouter({
         resource,
       };
     }),
-  update: adminProcedure.input(resourceCreateSchema).mutation(updateResource),
+  update: privateProcedure.input(resourceCreateSchema).mutation(updateResource),
```

Comment:
This change is a risky one. We need to add an extra safe-guard in this method to make sure that the user can edit the resource they are trying to edit.

Reply:
You are right, Ive pushed a commit that adds an `OwnerProcedure` for mutations like this that should be a step up from `privateProcedure`